Src/FluentAssertions.Json/ObjectAssertionsExtensions.cs

```diff
@@ -0,0 +1,97 @@
+        {
+            return BeJsonSerializable<object>(assertions, options => options, because, becauseArgs);
+        }
+
+        /// <summary>
+        /// Asserts that an object can be serialized and deserialized using the JSON serializer and that it stills retains
+        /// the values of all members.
+        /// </summary>
+        /// <param name="assertions"></param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion 
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <see cref="because" />.
+        /// </param>
+        public static AndConstraint<ObjectAssertions> BeJsonSerializable<T>(this ObjectAssertions assertions, string because = "", params object[] becauseArgs)
+        {
+            return BeJsonSerializable<T>(assertions, options => options, because, becauseArgs);
+        }
+
+        /// <summary>
+        /// Asserts that an object can be serialized and deserialized using the JSON serializer and that it stills retains
+        /// the values of all members.
+        /// </summary>
+        /// <param name="options"></param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion 
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <see cref="because" />.
+        /// </param>
+        /// <param name="assertions"></param>
+        public static AndConstraint<ObjectAssertions> BeJsonSerializable<T>(this ObjectAssertions assertions, Func<EquivalencyAssertionOptions<T>, EquivalencyAssertionOptions<T>> options, string because = "", params object[] becauseArgs)
+        {
+            try
+            {
+                var deserializedObject = CreateCloneUsingJsonSerializer(assertions.Subject);
+
+                var defaultOptions = AssertionOptions.CloneDefaults<T>()
+                    .RespectingRuntimeTypes()
+                    .IncludingFields()
+                    .IncludingProperties();
+
+                var typedSubject = (T)assertions.Subject;
+                ((T)deserializedObject).Should().BeEquivalentTo(typedSubject, _ => options(defaultOptions));
+            }
+            catch (Exception exc)
+            {
+                Execute.Assertion
+                    .BecauseOf(because, becauseArgs)
+                    .FailWith("Expected {0} to be JSON serializable{reason}, but serialization failed with:\r\n\r\n{1}",
```

Comment:
I recommend rewriting the failure as

`FailWith("Expected {object:context} to be JSON serializable{reason}, but serializing {0} failed with {1}", assertions.Subject, exc)`

This will make sure that the name of the variable on which you call `Should()` will be included in the message. But you need to add `[CustomAssertion]` to all your new methods. And by passing in the entire exception object, you let FA figure out how to format that the best way.

Reply:
Thanks!  I've updated the PR as suggested